posthog/async_migrations/migrations/0002_events_sample_by.py

```diff
@@ -68,134 +29,4 @@ class Migration(AsyncMigrationDefinition):
-                rollback=f"DETACH TABLE {EVENTS_TABLE_NAME}_mv ON CLUSTER '{CLICKHOUSE_CLUSTER}'",
-            ),
-            AsyncMigrationOperation(
-                fn=lambda _: set_instance_setting("COMPUTE_MATERIALIZED_COLUMNS_ENABLED", True),
-                rollback_fn=lambda _: set_instance_setting("COMPUTE_MATERIALIZED_COLUMNS_ENABLED", False),
-            ),
-            AsyncMigrationOperation(
-                fn=lambda query_id: run_optimize_table(
-                    unique_name="0002_events_sample_by", query_id=query_id, table_name=EVENTS_TABLE_NAME, final=True
-                )
-            ),
-        ]
-
-        _operations = create_table_op + old_partition_ops + detach_mv_ops + last_partition_op + post_insert_ops
-        return _operations
-
-    def is_required(self):
-        if settings.MULTI_TENANCY:
-            return False
-
-        table_engine = sync_execute(
-            "SELECT engine_full FROM system.tables WHERE database = %(database)s AND name = %(name)s",
-            {"database": settings.CLICKHOUSE_DATABASE, "name": EVENTS_TABLE},
-        )[0][0]
-
-        if "Distributed" in table_engine:
-            return False
-
-        return (
-            "ORDER BY (team_id, toDate(timestamp), event, cityHash64(distinct_id), cityHash64(uuid))"
-            not in table_engine
-        )
-
-    def precheck(self):
-        events_failed_table_exists = sync_execute(f"EXISTS {FAILED_EVENTS_TABLE_NAME}")[0][0]
-        if events_failed_table_exists:
-            return (
-                False,
-                f"{FAILED_EVENTS_TABLE_NAME} already exists. We use this table as a backup if the migration fails. You can delete or rename it and restart the migration.",
-            )
-
-        events_table = "sharded_events" if CLICKHOUSE_REPLICATION else "events"
-        return analyze_enough_disk_space_free_for_table(events_table, required_ratio=1.5)
-
-    def healthcheck(self):
-        result = sync_execute("SELECT free_space FROM system.disks")
-        # 100mb or less left
-        if int(result[0][0]) < 100000000:
-            return (False, "ClickHouse available storage below 100MB")
-
-        return (True, None)
-
-    @cached_property
-    def _partitions(self):
-        return list(
-            sorted(
-                row[0]
-                for row in sync_execute(
-                    f"SELECT DISTINCT toUInt32(partition) FROM system.parts WHERE database = %(database)s AND table='{EVENTS_TABLE}'",
-                    {"database": CLICKHOUSE_DATABASE},
-                )
-            )
-        )
-
-    def _events_table_engine(self) -> str:
-        rows = sync_execute(
-            "SELECT engine FROM system.tables WHERE database = %(database)s AND name = 'events'",
-            {"database": CLICKHOUSE_DATABASE},
-        )
-        return rows[0][0]
+    # Check older versions of the file for the migration code
```

Comment:
Does removing this code help us in any way? Is it actually removing complexity? 

In reality we won't look at this code again until someone on version 1.33.0 complains that something didn't work. At which point it becomes more annoying to debug.

I like the idea of keeping a record in of what backfills we ran. 

Unrelated to the core sentiment, we currently link to these files from the UI

Reply:
> Does removing this code help us in any way? Is it actually removing complexity?

Yes, tests can fail (& we're wasting resources running them) if we change some functions used in the code and we need to codemod these files too which has been annoying at least once.